astropy/table/table.py

```diff
@@ -3379,6 +3379,79 @@ def _replace_cols(self, columns):
 
         self.columns = columns
 
+    def setdefault(self, name, default):
+        """Ensure a column named ``name`` exists.
+
+        If ``name`` is already present then ``default`` is ignored.
+        Otherwise ``default`` can be any data object which is acceptable as
+        a `~astropy.table.Table` column object or can be converted.  This
+        includes mixin columns and scalar or length=1 objects which get
+        broadcast to match the table length.
+
+        Parameters
+        ----------
+        name : str
+            Name of the column.
+        default : object
+            Data object for the new column
```

Comment:
I don't feel guilty about this because I copy-pasted it from the docstring of `Table.add_column()`: https://github.com/astropy/astropy/blob/c813acac1715c4d0adc4004798e31b3ce8519353/astropy/table/table.py#L2345